external/kafka-0-10-sql/src/test/scala/org/apache/spark/sql/kafka010/KafkaContinuousSourceSuite.scala

```diff
@@ -212,13 +212,14 @@ class KafkaContinuousSourceTopicDeletionSuite extends KafkaContinuousTest {
         testUtils.createTopic(topic2, partitions = 5)
         eventually(timeout(streamingTimeout)) {
           assert(
-            query.lastExecution.executedPlan.collectFirst {
-              case scan: ContinuousScanExec
-                if scan.stream.isInstanceOf[KafkaContinuousStream] =>
-                scan.stream.asInstanceOf[KafkaContinuousStream]
-            }.exists { stream =>
-              // Ensure the new topic is present and the old topic is gone.
-              stream.knownPartitions.exists(_.topic == topic2)
+            query.lastExecution.executedPlan.exists {
```

Comment:
I guess this logic is changed?

suppose there are two `ContinuousScanExec` and the first one do not satisify `stream.knownPartitions.exists(_.topic == topic2)` but the second one satisify it.

Reply:
You are right. The logic has changed, [0031b96](https://github.com/apache/spark/pull/35833/commits/0031b96ae460fa0adf42ae14d0ea6c66b5df0501) revert this changed.

Thanks ~